lib/utils/ReporterStats.js

```diff
@@ -3,14 +3,14 @@ import crypto from 'crypto'
 import sanitize from '../helpers/sanitize'
 
 class RunnableStats {
-    constructor (type) {
+    constructor (type, start) {
```

Comment:
Can you elaborate why these changes are necessary?

Reply:
This is explained in a [previous comment](https://github.com/webdriverio/webdriverio/pull/2556#issuecomment-359717793) and has to do with incorrect timestamps of buffered messages.
> We've been running this patch for a couple of days now and the retries seem to be working nicely. I just noticed however that the reported duration of tests have dropped down to tens of millis. Only tests that have passed on their final run are correctly reporting duration. This correlates with the messages being buffered. I suspect start/end timestamps are assigned by the message handler as apposed to being contained within the message itself. Will look into this a bit later - should be an easy fix.